### PR TITLE
Precommits

### LINKED ISSUE #32: precommit hooks
**Is your feature request related to a problem? Please describe.**
Follow code standards

### PR DESCRIPTION
**Pull request by juanchaur**
Fixes: https://github.com/dominguezcelada/rent-flat-scraper/issues/32
screen shot 2018-07-13 at 16 39 15

### CODE REVIEW

**Comment by oscard0m** [comment]
@juanchaur try to use the PR template next time ;)

**Comment by oscard0m** [comment]
Tested `make` command and working properly:
```bash
                                                                                                              [00:07:10]
>>> >>> >>>  Install dependecies
npm i
npm WARN deprecated phantomjs@2.1.7: Package renamed to phantomjs-prebuilt. Please update 'phantomjs' package references to 'phantomjs-prebuilt'

> fsevents@1.2.4 install /Users/oscardom/dev/personal-projects/rent-flat-scraper/node_modules/fsevents
> node install

[fsevents] Success: "/Users/oscardom/dev/personal-projects/rent-flat-scraper/node_modules/fsevents/lib/binding/Release/node-v57-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile

> phantomjs@2.1.7 install /Users/oscardom/dev/personal-projects/rent-flat-scraper/node_modules/phantomjs
> node install.js

Considering PhantomJS found at /usr/local/bin/phantomjs
Found PhantomJS at /usr/local/bin/phantomjs ...verifying
PhantomJS detected, but wrong version 1.9.8 @ /usr/local/bin/phantomjs.
Downloading https://github.com/Medium/phantomjs/releases/download/v2.1.1//phantomjs-2.1.1-macosx.zip
Saving to /var/folders/j7/qxrlsvhs67j5ybd18l8xkyym0000gn/T/phantomjs/phantomjs-2.1.1-macosx.zip
Receiving...
  [======================================--] 95%
Received 16746K total.
Extracting zip contents
Removing /Users/oscardom/dev/personal-projects/rent-flat-scraper/node_modules/phantomjs/lib/phantom
Copying extracted folder /var/folders/j7/qxrlsvhs67j5ybd18l8xkyym0000gn/T/phantomjs/phantomjs-2.1.1-macosx.zip-extract-1531519663939/phantomjs-2.1.1-macosx -> /Users/oscardom/dev/personal-projects/rent-flat-scraper/node_modules/phantomjs/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /Users/oscardom/dev/personal-projects/rent-flat-scraper/node_modules/phantomjs/lib/phantom/bin/phantomjs

> nodemon@1.17.5 postinstall /Users/oscardom/dev/personal-projects/rent-flat-scraper/node_modules/nodemon
> node bin/postinstall || exit 0

Love nodemon? You can now support the project via the open collective:
 > https://opencollective.com/nodemon/donate

added 526 packages in 32.94s
```

After installing, the app with `--fotocasa` flag works. ✅

File: precommits/pre-commit [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	files=$(git diff --cached --name-only --diff-filter=AM)
+else
+	files=$@
+fi
+
+pattern="src.*\.ts$"
+status=0
+for file in $files
+do
+	if [[ ${file} =~ $pattern ]]; then
```
Comment (oscard0m) [inline precommits/pre-commit:15]: `index.ts` it's outside `src` folder. We should move it to a folder under src so everything is run under the hooks or change the pattern of the hooks to consider files outside that folder.

Comment (oscard0m) [reply precommits/pre-commit:15]: Example:
```bash
diff --git a/index.ts b/index.ts
index f885c83..af94c74 100644
--- a/index.ts
+++ b/index.ts
@@ -3,7 +3,7 @@ import { start } from "./server/start";
 start()
   .then(() => {
     // tslint:disable-next-line:no-console
-    console.log("Connected!");
+    console.log('Connected!');
   }).catch((err) => {
     // tslint:disable-next-line:no-console
     console.error(`Error starting server: ${err.message}`);
```
--------------------------------------------------------
Prehooks not jumping for this file!

```bash
gcmsg "test"                                                                                                                     [00:17:13]
[precommits 1ec2a8c] test
 1 file changed, 1 insertion(+), 1 deletion(-)
```

**Review by oscard0m — CHANGES_REQUESTED** [review]

File: precommits/pre-commit (hunk 1 above)
Comment (juanchaur) [reply precommits/pre-commit:15]: Correct. The index.ts isn’t included yet. I’ll do the change tonight

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+INIT_TASKS:=		create-precommit dependency-install
```
Comment (nerder) [inline Makefile:1]: I think we should vote the use of `Makefile` probably is a bit out of scope for the actual issue this PR solves.

File: package.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
     "start:ts": "./node_modules/.bin/ts-node index.ts",
     "watch": "./node_modules/.bin/nodemon -e ts --exec npm start",
     "lint": "./node_modules/.bin/tslint -c tslint.json 'src/**/*.ts'",
+    "precommitLint": "./node_modules/.bin/tslint -c tslint.json ",
```
Comment (nerder) [inline package.json:11]: i will not create it as a npm script, why not simply run it from the `pre-commit` script?

File: precommits/pre-commit [hunk 4]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (nerder) [inline precommits/pre-commit:1]: ### Statement
I think that a pre-commit is a bit too strict as a constraints, it's fine to commit non linted code IMO. 

### Example
I can imagine the use case of a partial commit that you will make on your staging area in order to fix a stable version of your code while developing and later on rebase and eventually `amend` that commit to give a better explaination to the reviewer, i don't see the needs in that case to have a that level of strictness.

### Proposed alternative solution

- `pre-push`

File: Makefile (hunk 2 above)
Comment (oscard0m) [reply Makefile:1]: What are the cons to not use it?

File: package.json (hunk 3 above)
Comment (oscard0m) [reply package.json:11]: It should be at the pre-commit stage but, to have the option in an `npm` is a `good-to-have` to me

File: precommits/pre-commit (hunk 4 above)
Comment (oscard0m) [reply precommits/pre-commit:1]: +1! It happened to me in other projects. it should be at a pre-push level so you can work regularly in intermediate commits avoiding the `--no-verify`.

File: Makefile (hunk 2 above)
Comment (nerder) [reply Makefile:1]: Is just not useful, npm scripts are able to perform the same.

**Comment by oscard0m** [comment]
Closed due to inactivity

**Pull request closed without merging**